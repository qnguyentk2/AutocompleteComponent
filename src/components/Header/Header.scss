// @import '@assets/styles/colors.scss';
@import '../../../assets/styles/colors.scss';

.Header {
  position: absolute;
  top: 0;
  width: -webkit-fill-available;
  // background-color: red;
  display: flex;
  padding: 24px 26px 24px 26px;
  justify-content: space-between;
  z-index: 999 !important;
}

.leftSide {
  // background-color: yellow;
  overflow: hidden;
}

.leftSide a {
  float: left;
  color: $menu-item-text-color;
  text-align: center;
  padding: 0 16px;
  text-decoration: none;
  font-size: 14px;
  height: 16px;
  line-height: 32px;
  cursor: pointer;

  &.themeMode {
    margin-top: 5px;
  }
}