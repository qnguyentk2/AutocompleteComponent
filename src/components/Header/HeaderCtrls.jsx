import React, { useContext } from 'react';
import { Link } from 'react-router-dom';
import GlobalContext from '@contexts/GlobalContext';
import RouteLinks from '@const/routes';
import BurgerMenu from '@components/BurgerMenu/BurgerMenu';
import lightMode from '@assets/imgs/icons/light-mode.svg';
import darkMode from '@assets/imgs/icons/dark-mode.svg';
import styles from './Header.scss';

const HeaderControls = () => {
  const { isLightMode, handleChangeThemeMode } = useContext(GlobalContext);
  const menuItemArr = ['Become a host', 'Help', 'Sign up', 'Log in'];

  const renderChangeMode = () => {
    return (
      <a
        onClick={handleChangeThemeMode}
        className={styles.themeMode}
      >
        <img alt="theme" src={isLightMode ? lightMode : darkMode} />
      </a>
    )
  }

  const renderMenuMobileMode = () => {
    return (
      <>
        {renderChangeMode()}
        <BurgerMenu />
      </>
    )
  }

  const renderMenuDestopMode = () => (
    <div className={styles.leftSide}>
      {renderChangeMode()}
      {menuItemArr.map(item => (
        <Link key={item} to={`${RouteLinks.becomeAHost.path}`}>{item}</Link>
      ))}
    </div>
  )

  return {
    renderMenuMobileMode,
    renderMenuDestopMode
  }
}

export default HeaderControls;