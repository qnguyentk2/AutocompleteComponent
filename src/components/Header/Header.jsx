import React from 'react';
import { withRouter, Link } from 'react-router-dom';
import RouteLinks from '@const/routes';
import { isMobileOnly } from 'react-device-detect';
import logo from '@assets/imgs/icons/logo.svg';
import styles from './Header.scss';
import HeaderCtrls from './HeaderCtrls';

const Header = props => {
  const { renderMenuMobileMode, renderMenuDestopMode } = HeaderCtrls(props);
  return (
    <div className={styles.Header}>
      <div className={styles.rightSide}>
        <Link to={`${RouteLinks.landingPage.path}`}>
          <img className={styles.logo} alt="logo" src={logo} />
        </Link>
      </div>
      {
        isMobileOnly ? renderMenuMobileMode() : renderMenuDestopMode()
      }
    </div>
  );
};

export default withRouter(Header);
