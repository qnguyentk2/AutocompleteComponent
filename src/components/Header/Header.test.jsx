/* eslint-disable no-undef */
import React from 'react';
// import { renderHook } from '@testing-library/react-hooks';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Header from './Header';

const setup = () => {
  const wrapper = mount(<Header />);

  return { wrapper };
}

describe('Test App', () => {
  test('renders without crashing', () => {
    const { wrapper } = setup();
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});