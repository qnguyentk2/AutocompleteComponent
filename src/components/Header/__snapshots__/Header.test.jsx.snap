// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App renders without crashing 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
